### PR TITLE
pre-commit update

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR replaces #4203 to update the `pre-commit` version of `isort`, but is based on the working version of `main` for `isort` within `nox` (see #4202).

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Comment by trexfeathers** [comment]
This definitely touches the same lines as #4181 so I suggest we wait until that gets merged to avoid confusion.

**Comment by bjlittle** [comment]
@trexfeathers To be honest, I'm happy for this to go in first and I'll rebase #4181 :+1:

**Review by trexfeathers — APPROVED** [review]

**Pull request merged**
